be/src/storage/tablet_manager.cpp

```diff
@@ -1018,12 +1018,23 @@ Status TabletManager::start_trash_sweep() {
                 TabletSharedPtr& tablet = *it;
                 std::string tablet_path = tablet->tablet_path();
                 if (Env::Default()->path_exists(tablet_path).ok()) {
-                    // take snapshot of tablet meta
-                    std::string meta_file_path = path_util::join_path_segments(
-                            tablet->tablet_path(), std::to_string(tablet->tablet_id()) + ".hdr");
-                    tablet->tablet_meta()->save(meta_file_path);
-                    OLAPStatus rm_st = move_to_trash(tablet_path, tablet_path);
-                    if (rm_st == OLAP_SUCCESS) {
+                    if (tablet->keys_type() == KeysType::PRIMARY_KEYS) {
+                        StatusOr<std::string> st = SnapshotManager::instance()->snapshot_trash(tablet, 0, 180);
```

Comment:
This doesn't make sense, the purpose of this part of the code is to write meta `snapshot` to the tablet path. 
Maybe you can rename snapshot_trash to write_meta_snapshot(tablet, tablet_path + "/meta.hr") and implement it